app/models/rbac.rb

```diff
@@ -324,9 +347,19 @@ def self.get_user_info(user, userid, miq_group, miq_group_id)
 
   def self.filtered(objects, options = {})
     if objects.present?
-      Rbac.search(options.merge(:targets => objects, :results_format => :objects)).first
+      Rbac.search(options.reverse_merge(:targets => objects, :results_format => :objects, :skip_counts => true)).first
     else
-      objects
+      # when the present is removed, then this case statement can be removed
+      case options[:results_format]
```

Comment:
`:results_format` is a travesty, and one half of the variable behaviour `filtered` was intended to kill off.

:-1: